libraries/anvill_passes/include/anvill/IndirectJumpPass.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <anvill/ABI.h>
+#include <llvm/IR/InstIterator.h>
+#include <llvm/IR/Instructions.h>
+#include <llvm/Pass.h>
+
+namespace anvill {
+namespace {
+static bool isTargetInstrinsic(const llvm::CallInst *callinsn) {
+  if (const auto *callee = callinsn->getCalledFunction()) {
+    return callee->getName().equals(kAnvillSwitchCompleteFunc);
```

Comment:
So this is an interesting question and I meant to bring it up in the PR. Yes because the range for the index is conservative, the analysis will always consider all possible loads from the offset table. Now if the switch is truly incomplete (ie. we were not able to recover all possible jump targets) then the pcbinding lookup in switch lowering should fail because we won't have a label for one of the outgoing program counters. So in that sense I think the switch recovery pass (the actual switch generation) will only succeed for mislabeled incomplete switch intrinsics which then begs the question, should we even attempt to run that pass on the incomplete intrinsic? That being said the JumpTableAnalysis results will still be valid, even if the intrinsic is incomplete. Perhaps the best solution is to apply the analysis to both types of intrinsics and only run the lowering pass for complete switches? The assumption being that for incomplete switches we need to pass back that jump table analysis to the lifter itself.